src/store/modules/nutrition/getters.ts

```diff
@@ -1,7 +1,45 @@
 import { GetterTree } from 'vuex';
 
-export const getters: GetterTree<any, any> = {
-    cachedDailyIntake: (state: any) => (day: string) => {
+import { RootState, NutritionState } from '@/models/store';
+import { FoodInfo, FoodNutrient, Meal } from '@/models/nutrition';
+
+export const getters: GetterTree<NutritionState, RootState> = {
+    cachedDailyIntake: (state: NutritionState) => (day: string) => {
         return state.intakes[day];
     },
+    foodInfo: (state: NutritionState) => (id: string): FoodInfo | undefined => {
+        return state.foodInfoArray.length > 0
+            ? state.foodInfoArray.find((foodInfo: FoodInfo) => foodInfo.resource_id === id)
+            : undefined;
+    },
+    getMealFromCurrentDay: (state: NutritionState) => (mealName: string): Meal | null => {
+        const mealIndex = state.currentIntake.meals && state.currentIntake.meals
+            .findIndex((mealItem: Meal) => mealItem.name.toLowerCase() === mealName.toLowerCase());
+        if (typeof mealIndex === 'number') {
+            return state.currentIntake.meals[mealIndex];
+        }
+        return null;
+    },
+    foodNutrients: (
+            state: NutritionState,
+            getters: any) =>
+            (id: string): {[key: string]: any} | null => {
+                const foodInfo = getters.foodInfo(id);
+                if (!foodInfo) { return null; }
```

Comment:
In generall I don't use one-line IF statements. It saves only two lines but, in my opinion, at the cost of readability.

Reply:
I will consider this idea

Reply:
Done